### DISCUSSION TITLE
ESP32 based smart plug powers off to early because of to high measured internal temperature sensor?

### DISCUSSION THREAD

**Discussion by CosmicDuck** [arendst/Tasmota#22633] [Q&A] [answered]
I use my smart plugs (four for the moment) with tasmota to control my electical heaters.
(ASIN B0CHMMKZCQ)
If the power consuming of the heaters is to high after some time the smart meters power off the relais.
I found in the debug logs that the measured internal temperature reaches the over 90 ° C limit and powers the switch off. 

```
11:18:37.725 NRG: Temperature 90.1
11:18:37.726 SRC: Overtemp
11:18:37.732 MQT: stat/MeinSachsHeizluefter/RESULT = {"POWER":"OFF"}
11:18:37.735 MQT: stat/MeinSachsHeizluefter/POWER = OFF
11:18:42.816 CFG: Saved, Count 72, Bytes 4096
```
Generally this is a good safety function.

By the way without consuming power the internal sensor allready is measuring round about + 50 K higher than the environment temperature (Environment is < 7 °C, ESP32 is > 60 °)

I didn't found a solution yet to change the power off limit of 90 °C so in my opinion maybe the limit is a little to high for a safety function of the plug if only the esp32 internal sensor registered a higher temperature.
I only found in the review of the product a hint how I can completly disable the esp32 sensor:

```
Die einzige Möglichkeit dieses Verhalten abzustellen ist: über die Konsole mit
"SetOption146 0" die Telemetrie des Sensors zu stoppen und danach
"SetSensor127 0" die Auswertung des Sensors komplett zu unterbinden.
```

By the way I think it is better to change the limit or something else or maybe to change the calibration of the internal sensor.
I will recheck the internal temperature after plugging the device to check if it could be a problem of the the esp32 temperature calibration itself.

**Comment by sfromis** [comment, marked as answer]
The over-temperature threshold can be selected using `SetOption42`.

**Reply by CosmicDuck** [reply]
Thank you. So I can add a higher limit before the relais will go off.
By the way I checked the cold start of the tasmota device and found out that the temperature difference is 21 °C - °C7= 14 K after 5 s of fresh plugin the smart device.
So maybe the calibration is not very good and has only a difference of 10 K in cold state.
After warmup of 10 minutes in a warming area of 20 °C the esp32 temperature is at 57 °C and a temperature difference of 37 K with power off relais.
With power on relais without a load the temperature increased again to a smaller gap to 60 °C - 20 °C = 40 K.
So with this value it is save to increase the limit to 70 °C + 40 K = 110 °C. Maybe 100 °C are allready enough for my 2000 W heater although the device is build for 16 A. 70 °C because cables should not get hotter if I remember correctly.

**Reply by CosmicDuck** [reply]
Here some links now found for further reading:
[https://tasmota.github.io/docs/Commands/#setoption42](https://tasmota.github.io/docs/Commands/#setoption42)

**Reply by sfromis** [reply]
If your aim is to find a sensible point for over-temperature, do not assume the ESP32-C3 *internal* chip temperature to even be a good proxy for device temperature, and do not assume that the gap is constant or linear with increasing temperature. I'd not even recommend enabling `SetOption146`
